eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -594,42 +590,68 @@ function getOperationScope(path: string): ResourceScope {
   return ResourceScope.Tenant; // all the templates work as if there is a tenant decorator when there is no such decorator
 }
 
-function addNonResourceMethodDecorators(
-  codeModel: CodeModel,
-  metadata: NonResourceMethod[]
-) {
-  codeModel.clients[0].decorators ??= [];
-  codeModel.clients[0].decorators.push({
-    name: nonResourceMethodMetadata,
-    arguments: convertMethodMetadataToArguments(metadata)
-  });
-}
+/**
+ * Builds the unified ARM provider schema from resource metadata and non-resource methods.
+ * @param resourceModels - All resource models in the code model
+ * @param resourceModelToMetadataMap - Map of resource model IDs to their metadata
+ * @param nonResourceMethods - Map of non-resource method IDs to their metadata
+ * @param sdkContext - The emitter context for logging
+ * @returns The complete ARM provider schema
+ */
+function buildArmProviderSchema(
+  resourceModels: InputModelType[],
+  resourceModelToMetadataMap: Map<string, ResourceMetadata>,
+  nonResourceMethods: Map<string, NonResourceMethod>,
+  sdkContext: CSharpEmitterContext
```

Comment:
please keep in mind that the `sdkContext` should always be the first parameter in all functions.